packages/amplify-provider-awscloudformation/src/utils/remove-dependent-function.ts

```diff
@@ -0,0 +1,72 @@
+import { $TSAny, $TSContext, $TSObject, pathManager } from 'amplify-cli-core';
+import path from 'path';
+import * as TransformPackage from 'graphql-transformer-core';
+
+export async function removeDependencyOnFunctions(
+  context: $TSContext,
+  apiResource: $TSObject,
+  allResources: $TSObject[],
+): Promise<$TSObject[]> {
+  // get #current-cloud-backed and cloud backend schema.graphql
+  let functionResource;
+  let functionResourceNames;
+  const backendDir = pathManager.getBackendDirPath();
+  const currentBackendDir = pathManager.getCurrentCloudBackendDirPath();
+  const modelsDeleted = await getSchemaDiff(currentBackendDir, backendDir, apiResource[0].resourceName);
+  if (modelsDeleted.length) {
+    functionResourceNames = await context.amplify.invokePluginMethod(context, 'function', undefined, 'getDependentFunctions', [
+      context,
+      allResources,
+      backendDir,
+      modelsDeleted,
+    ]);
+    if (functionResourceNames.length) {
+      context.print.info('');
+
+      context.print.warning(`Functions ${functionResourceNames} have access to a removed GraphQL API model(s) ${modelsDeleted}`);
+
+      let continueToPush = context.exeInfo && context.exeInfo.inputParams && context.exeInfo.inputParams.yes;
+      let forcePush = context?.exeInfo?.forcePush;
+
+      if (!continueToPush && !forcePush) {
+        continueToPush = await context.amplify.confirmPrompt(
+          'Do you want to remove the GraphQL model access on these affected functions?',
```

Comment:
```suggestion
          'Do you want to automatically remove the GraphQL model(s) access from the function(s)? (Note: This may cause runtime failures if your function(s) are expected access to the removed model(s).)',
```

Reply:
Changing to this , my test is getting stuck in nexpect. I have also changed it in  tests also.